src/app/clusters/mode-select-server/mode-select-server.cpp

```diff
@@ -109,35 +114,201 @@ CHIP_ERROR ModeSelectAttrAccess::Read(const ConcreteReadAttributePath & aPath, A
-                                                  const ModeSelect::Commands::ChangeToMode::DecodableType & commandData)
+Status ChangeToMode(EndpointId endpointId, uint8_t newMode)
 {
     MATTER_TRACE_SCOPE("ChangeToMode", "ModeSelect");
     ChipLogProgress(Zcl, "ModeSelect: Entering emberAfModeSelectClusterChangeToModeCallback");
-    EndpointId endpointId = commandPath.mEndpointId;
-    uint8_t newMode       = commandData.newMode;
+
     // Check that the newMode matches one of the supported options
     const ModeSelect::Structs::ModeOptionStruct::Type * modeOptionPtr;
     const ModeSelect::SupportedModesManager * gSupportedModeManager = ModeSelect::getSupportedModesManager();
     if (gSupportedModeManager == nullptr)
     {
         ChipLogError(Zcl, "ModeSelect: SupportedModesManager is NULL");
-        commandHandler->AddStatus(commandPath, Status::Failure);
-        return true;
+        return Status::Failure;
     }
     Status checkSupportedModeStatus = gSupportedModeManager->getModeOptionByMode(endpointId, newMode, &modeOptionPtr);
     if (Status::Success != checkSupportedModeStatus)
     {
         ChipLogProgress(Zcl, "ModeSelect: Failed to find the option with mode %u", newMode);
-        commandHandler->AddStatus(commandPath, checkSupportedModeStatus);
-        return true;
+        return checkSupportedModeStatus;
     }
     ModeSelect::Attributes::CurrentMode::Set(endpointId, newMode);
 
+    return Status::Success;
+}
+
+} // anonymous namespace
+
+#if defined(MATTER_DM_PLUGIN_SCENES_MANAGEMENT) && CHIP_CONFIG_SCENES_USE_DEFAULT_HANDLERS
+static constexpr size_t kModeSelectMaxEnpointCount =
+    MATTER_DM_MODE_SELECT_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+
+static void timerCallback(System::Layer *, void * callbackContext);
+static void sceneModeSelectCallback(EndpointId endpoint);
+using ModeSelectEndPointPair = scenes::DefaultSceneHandlerImpl::EndpointStatePair<uint8_t>;
+using ModeSelectTransitionTimeInterface =
+    scenes::DefaultSceneHandlerImpl::TransitionTimeInterface<kModeSelectMaxEnpointCount,
+                                                             MATTER_DM_MODE_SELECT_CLUSTER_SERVER_ENDPOINT_COUNT>;
+
+class DefaultModeSelectSceneHandler : public scenes::DefaultSceneHandlerImpl
+{
+public:
+    DefaultSceneHandlerImpl::StatePairBuffer<uint8_t, kModeSelectMaxEnpointCount> mSceneEndpointStatePairs;
+    // As per spec, 1 attribute is scenable in the mode select cluster
+    static constexpr uint8_t scenableAttributeCount = 1;
+
+    DefaultModeSelectSceneHandler() = default;
+    ~DefaultModeSelectSceneHandler() override {}
+
+    // Default function for the mode select cluster, only puts the mode select cluster ID in the span if supported on the given
+    // endpoint
+    virtual void GetSupportedClusters(EndpointId endpoint, Span<ClusterId> & clusterBuffer) override
+    {
+        ClusterId * buffer = clusterBuffer.data();
```

Comment:
you can move this below. Also you probably can do directly `clusterBuffer[0] = ...` without referencing data.